rust-src/concordium_base/src/transactions.rs

```diff
@@ -0,0 +1,2733 @@
+        let payload = Payload::TransferToEncrypted { amount };
+        make_transaction(
+            sender,
+            nonce,
+            expiry,
+            GivenEnergy::Add {
+                num_sigs,
+                energy: cost::TRANSFER_TO_ENCRYPTED,
+            },
+            payload,
+        )
+    }
+
+    /// Transfer the given amount from encrypted to public balance of the given
+    /// account. The payload may be constructed using
+    /// [encrypted_transfers::make_sec_to_pub_transfer_data]
+    pub fn transfer_to_public(
+        num_sigs: u32,
+        sender: AccountAddress,
+        nonce: Nonce,
+        expiry: TransactionTime,
+        data: SecToPubAmountTransferData<EncryptedAmountsCurve>,
+    ) -> PreAccountTransaction {
+        // FIXME: This payload could be returned as well since it is only borrowed.
+        let payload = Payload::TransferToPublic {
+            data: Box::new(data),
+        };
+        make_transaction(
+            sender,
+            nonce,
+            expiry,
+            GivenEnergy::Add {
+                num_sigs,
+                energy: cost::TRANSFER_TO_PUBLIC,
+            },
+            payload,
+        )
+    }
+
+    /// Construct a transfer with schedule transaction, sending to the given
+    /// account.
+    pub fn transfer_with_schedule(
+        num_sigs: u32,
+        sender: AccountAddress,
+        nonce: Nonce,
+        expiry: TransactionTime,
+        receiver: AccountAddress,
+        schedule: Vec<(Timestamp, Amount)>,
+    ) -> PreAccountTransaction {
+        let num_releases = schedule.len() as u16;
+        let payload = Payload::TransferWithSchedule {
+            to: receiver,
+            schedule,
+        };
+        make_transaction(
+            sender,
+            nonce,
+            expiry,
+            GivenEnergy::Add {
+                num_sigs,
+                energy: cost::scheduled_transfer(num_releases),
+            },
+            payload,
+        )
+    }
+
+    /// Construct a transfer with schedule and memo transaction, sending to the
+    /// given account.
+    pub fn transfer_with_schedule_and_memo(
+        num_sigs: u32,
+        sender: AccountAddress,
+        nonce: Nonce,
+        expiry: TransactionTime,
+        receiver: AccountAddress,
+        schedule: Vec<(Timestamp, Amount)>,
+        memo: Memo,
+    ) -> PreAccountTransaction {
+        let num_releases = schedule.len() as u16;
+        let payload = Payload::TransferWithScheduleAndMemo {
+            to: receiver,
+            memo,
+            schedule,
+        };
+        make_transaction(
+            sender,
+            nonce,
+            expiry,
+            GivenEnergy::Add {
+                num_sigs,
+                energy: cost::scheduled_transfer(num_releases),
+            },
+            payload,
+        )
+    }
+
+    /// Register the sender account as a baker.
+    /// TODO: Make a function for constructing the keys payload, with correct
```

Comment:
todo

Reply:
This already exists, so todo is removed.